build.gradle.kts

```diff
@@ -1,41 +1,37 @@
 import org.gradle.api.tasks.testing.logging.TestExceptionFormat
 import org.gradle.api.tasks.testing.logging.TestLogEvent
+import org.jetbrains.kotlin.gradle.dsl.ExplicitApiMode
 import java.io.ByteArrayOutputStream
 import java.io.FileOutputStream
 import java.util.Properties
 
 plugins {
-    kotlin("jvm") version "1.6.20"
+    kotlin("jvm") version "1.9.20"
     application
     `maven-publish`
+    id("org.jlleitschuh.gradle.ktlint") version "12.2.0"
+    id("org.jetbrains.kotlinx.binary-compatibility-validator") version "0.17.0"
 }
 
 val properties = "$buildDir/properties"
 
 object Versions {
     // Language
-    const val kotlin = "1.6.20"
-    const val kotlinLanguage = "1.6"
-    const val kotlinApi = "1.6"
-    const val jvmTarget = "1.8"
+    const val KOTLIN = "1.9.20"
+    const val KOTLIN_LANGUAGE = "1.9"
+    const val KOTLIN_API = "1.9"
+    const val JVM_TARGET = "1.8"
 
     // Deps
-    const val guava = "31.1-jre"
-    const val jansi = "2.4.0"
-    const val jline = "3.21.0"
-    const val junit5 = "5.9.3"
-    const val picoCli = "4.7.0"
-    const val partiql = "0.14.9"
+    const val JUNIT_5 = "5.9.3"
+    const val PARTIQL = "1.2.1"
```

Comment:
self-review: updated Scribe to use PLK `v1.2.1`, which includes the fixes from https://github.com/partiql/partiql-lang-kotlin/pull/1749